files/image_config/platform/rc.local

```diff
@@ -76,6 +77,57 @@ if [ -f /host/image-$sonic_version/platform/firsttime ]; then
         dpkg -i /host/image-$sonic_version/platform/$platform/*.deb
     fi
 
+    # If the unit booted into SONiC from another NOS's grub, we now
+    # install a grub for SONiC based on ONIE grub.
+    if [ -n "$onie_platform" ] && [ -n "$migration" ]; then
+        mkdir -p /host/image-$sonic_version/tmp
+        cd /host/image-$sonic_version/tmp
+
+        # Extract ONIE's ramdisk which contain's the ONIE grub
+        mount $onie_dev /mnt/onie-boot
+        cp /mnt/onie-boot/onie/initrd.img-3.2.69-onie initrd.img-3.2.69-onie.xz
```

Comment:
>.img-3.2.69-onie initrd.img-3.2.69- [](start = 37, length = 35)

hardcoding onie kernel version does not sound a good design. this puts the sonic installation process highly dependent on the onie installer we have in the box. In the future, you may ship a ftos box with newer onie, then this will break.

Reply:
for example, on my box the onie is initrd.img-3.2.35-onie